portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,33 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    private  ArrayList<String> messages;
+
+    @Override
+    public void init(){
+    messages = new ArrayList<>();
+    messages.add("Howdy! Howdy! Howdy!");
+    messages.add("Purple Puppy Chow");
+    messages.add("Shrimp! Heaven! Now!");
+    }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Leah!</h1>");
+    // Convert ArrayList to JSON
+    response.setContentType("application/json;");
+    response.getWriter().println(convertToJsonWithGSon(messages));
+  }
+  private String convertToJsonWithGSon(ArrayList messages) {
```

Comment:
Blank line between the functions. Also, great job on breaking out the conversion to a new method! This is a style of programming called SOLID. [Here](https://en.wikipedia.org/wiki/SOLID) is a good source on that.